RELEASE_NOTES.rst

```diff
@@ -21,6 +21,43 @@
 
 .. towncrier release notes start
 
+Airflow 2.10.1 (2024-08-30)
```

Comment:
Don't forget to update this date!

Reply:
Updated it to 2024-09-05, assuming we are doing rc1 today.